sdk/communication/Azure.Communication.Common/tests/Identity/EntraTokenCredentialTest.cs

```diff
@@ -0,0 +1,211 @@
+                _scopes));
+
+            Assert.Throws<ArgumentException>(() => new EntraCommunicationTokenCredentialOptions(
+                "",
+                _mockTokenCredential.Object,
+                _scopes));
+
+            Assert.Throws<ArgumentNullException>(() => new EntraCommunicationTokenCredentialOptions(
+                _resourceEndpoint,
+                null,
+                _scopes));
+
+            Assert.Throws<ArgumentException>(() => new EntraCommunicationTokenCredentialOptions(
+                _resourceEndpoint,
+                _mockTokenCredential.Object,
+                new string[] { }));
+
+            Assert.Throws<ArgumentNullException>(() => new EntraCommunicationTokenCredentialOptions(
+                _resourceEndpoint,
+                _mockTokenCredential.Object,
+                null));
+        }
+
+        [Test]
+        public void EntraTokenCredential_InitWhtScopes_InitWithDefaultScope()
+        {
+            var credential = new EntraCommunicationTokenCredentialOptions(
+                _resourceEndpoint,
+                _mockTokenCredential.Object);
+            var scopes = new[] { "https://communication.azure.com/clients/.default" };
+            Assert.AreEqual(credential.Scopes, scopes);
+        }
+
+        [Test]
+        public void EntraTokenCredential_Init_FetchesTokenImmediately()
+        {
+            // Arrange
+            var expiryTime = DateTimeOffset.Parse(SampleTokenExpiry, null, System.Globalization.DateTimeStyles.RoundtripKind);
+            var options = CreateEntraTokenCredentialOptions();
+            var mockTransport = CreateMockTransport(new[] { CreateMockResponse(200, TokenResponse) });
+            var entraTokenCredential = new EntraTokenCredential(options, mockTransport);
+            // Assert
+            _mockTokenCredential.Verify(tc => tc.GetTokenAsync(It.IsAny<TokenRequestContext>(), It.IsAny<CancellationToken>()), Times.Once);
+        }
+
+        [Test]
+        public async Task EntraTokenCredential_GetToken_ReturnsToken()
+        {
+            // Arrange
+            var expiryTime = DateTimeOffset.Parse(SampleTokenExpiry, null, System.Globalization.DateTimeStyles.RoundtripKind);
+            var options = CreateEntraTokenCredentialOptions();
+            var mockTransport = CreateMockTransport(new[] { CreateMockResponse(200, TokenResponse) });
+            var entraTokenCredential = new EntraTokenCredential(options, mockTransport);
+
+            // Act
+            var token = await entraTokenCredential.GetTokenAsync(CancellationToken.None);
+
+            // Assert
+            Assert.AreEqual(SampleToken, token.Token);
+            Assert.AreEqual(token.ExpiresOn, expiryTime);
+            _mockTokenCredential.Verify(tc => tc.GetTokenAsync(It.IsAny<TokenRequestContext>(), It.IsAny<CancellationToken>()), Times.Once);
+        }
+
+        [Test]
+        public async Task EntraTokenCredential_GetToken_MultipleCallsReturnsCachedToken()
```

Comment:
Can we add a `EntraTokenCredential_GetToken_InternalEntraTokenChangeInvalidatesCachedToken` where we implement the `TokenCredential` in a way that it can be configured to return a different token? See my JS comment here: https://github.com/Azure/azure-sdk-for-js/pull/31291#discussion_r1796936348

Reply:
added